File: java_app/src/main/java/com/petnagy/java/example/injection/ApplicationComponent.java
package com.petnagy.java.example.injection;

import com.petnagy.java.example.ProjectApplication;
import com.petnagy.java.example.redux.AppState;
import com.petnagy.koredux.Store;

import javax.inject.Singleton;

import dagger.Component;
import dagger.android.AndroidInjector;
import dagger.android.support.AndroidSupportInjectionModule;

@Singleton
@Component(modules = {AndroidSupportInjectionModule.class, ApplicationModule.class, DaggerActivityModule.class})
public interface ApplicationComponent extends AndroidInjector<ProjectApplication> {

    @Component.Builder
    abstract class Builder extends AndroidInjector.Builder<ProjectApplication> {
    }

    Store<AppState> exposeStore();
}

File: app/src/main/java/com/petnagy/app/injection/ApplicationComponent.kt
package com.petnagy.app.injection

import com.petnagy.app.ProjectApplication
import com.petnagy.app.redux.AppState
import com.petnagy.koredux.Store
import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule
import javax.inject.Singleton


@Singleton
@Component(modules = [(AndroidSupportInjectionModule::class), (ApplicationModule::class), (DaggerActivityModule::class)])
interface ApplicationComponent : AndroidInjector<ProjectApplication> {

    @Component.Builder
    abstract class Builder : AndroidInjector.Builder<ProjectApplication>()

    fun exposeStore(): Store<AppState>
}